app/models/manageiq/providers/redhat/infra_manager/refresh/parse/strategies/vm_inventory.rb

```diff
@@ -153,27 +153,58 @@ def vm_inv_to_guest_device_hashes(inv, lan_uids)
     end
 
     def vm_inv_to_network_hashes(inv, guest_device_uids)
-      reported_device = inv.respond_to?(:reported_devices) ? inv.reported_devices[0] : nil
-      inv_net = reported_device.ips if reported_device
-      result = []
-      return result if inv_net.nil?
+      reported_devices = inv.respond_to?(:reported_devices) ? inv.reported_devices : nil
+      all_networks = []
+      return all_networks if !reported_devices || reported_devices.blank?
+
+      network_by_mac = {}
+      host_name = inv.fqdn if inv.respond_to?(:fqdn)
+      reported_devices.to_miq_a.each do |reported_device|
+        inv_net = reported_device.ips
+        next if inv_net.nil?
 
-      inv_net.to_miq_a.each do |data|
-        new_result = { :ipaddress => data.address }
-        result << new_result unless new_result.blank?
+        ipaddresses = get_ipaddresses(inv_net)
+
+        ipaddresses.each do |ipv4address, ipv6address|
+          new_network = {
+            :hostname => host_name
+          }
+          new_network[:ipaddress] = ipv4address unless ipv4address.nil?
+          new_network[:ipv6address] = ipv6address unless ipv6address.nil?
+
+          device_mac = reported_device.mac ? reported_device.mac.address : nil
```

Comment:
can be:
device_mac = reported_device&.mac&.address

Reply:
As I see in the development setup guide the minimum required ruby version is 2.3.1+, and '&.' is not supported there. Is the guide outdated and the min required version was changed?

Reply:
done